packages/shared/ui/scripts/storybooks.ts

```diff
@@ -0,0 +1,84 @@
+import { spawn, ChildProcess } from 'child_process';
+import { readdirSync, statSync } from 'fs';
+import { join, dirname } from 'path';
+import { fileURLToPath } from 'url';
+
+const startStorybook = (name: string, command: string, dir: string): Promise<void> => {
+    return new Promise((resolve, reject) => {
+        const [cmd, ...args] = command.split(' ');
+        const storybookProcess: ChildProcess = spawn(cmd, args, { cwd: dir, env: { ...process.env }, shell: true });
+
+        storybookProcess.stdout?.on('data', (data) => {
+            const output = data.toString();
+            console.log(`${name}: ${output}`);
+
+            if (output.includes(`started`)) {
+                resolve();
+            }
+        });
+
+        storybookProcess.stderr?.on('data', (data) => {
+            console.error(`${name} stderr:`, data.toString());
+        });
+
+        storybookProcess.on('error', (error) => {
+            console.error(`${name} failed:`, error);
+            reject(error);
+        });
+
+        storybookProcess.on('close', (code) => {
+            if (code !== 0) {
+                console.error(`${name} exited with code ${code}`);
+            }
+        });
+    });
+};
+
+const findStorybookDirs = (baseDir: string): string[] => {
+    return readdirSync(baseDir)
+        .map((name) => join(baseDir, name))
+        .filter((dir) => statSync(dir).isDirectory() && readdirSync(dir).includes('.storybook'));
+};
+
+const startAllStorybooks = async () => {
+    const __filename = fileURLToPath(import.meta.url);
+    const __dirname = dirname(__filename);
+
+    const baseDir = join(__dirname, '../../../');
+    const storybookDirs = findStorybookDirs(baseDir);
+
+    console.log(`Found Storybooks in the following directories: ${storybookDirs.join(', ')}`);
+
+    let port = 6007;
```

Comment:
you ok with explicit port references and console logs being added here?

Reply:
there's a couple of other port references in other files too

Reply:
Yep, used for composite storybook